packages/web/components/Player/Box/PlayerGallery.tsx

```diff
@@ -1,63 +1,149 @@
+          fontFamily="heading"
+          mt={3}
+          mb={3}
+          casing="uppercase"
+          display="inline-block"
+          style={{ wordWrap: 'break-word', wordBreak: 'break-all' }}
+        >
+          {nft.title}
+        </Text>
+        <Text fontSize="sm">
+          {nft.priceInUsd && nft.priceInEth
+            ? `${nft.priceInEth}Ξ ($${nft.priceInUsd.toFixed(2)})`
+            : ``}
+        </Text>
+      </Flex>
+    </HStack>
+  </Link>
+);
+
+export const PlayerGallery: React.FC<Props> = ({ player, setRemoveBox }) => {
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const { favorites, data, loading } = useOpenSeaCollectibles({ player });
   return (
     <PlayerBox title="Gallery" setRemoveBox={setRemoveBox}>
-      {(fakeData || []).slice(0, show ? 999 : 3).map((nft) => (
-        <HStack alignItems="end" mb={6}>
-          <Flex width="126px" height="126px" mr={6}>
-            <Box
-              bgImage={`url(${nft.img})`}
-              backgroundSize="contain"
-              backgroundRepeat="no-repeat"
-              backgroundPosition="center"
-              width="124px"
-              height="124px"
-              m="auto"
-            />
-          </Flex>
-          <Box>
-            <Text
-              fontSize="xs"
-              fontFamily="heading"
-              mt={3}
-              mb={3}
-              casing="uppercase"
-            >
-              {nft.title}
-            </Text>
-            <Text fontSize="sm" mb="1">
-              {nft.priceInEth}Ξ ($
-              {parseFloat(`${nft.priceInEth * fakeEthPrice}`).toFixed(2)})
-            </Text>
-          </Box>
-        </HStack>
-      ))}
-      {(fakeData || []).length > 3 && (
+      {!loading &&
+        favorites &&
+        favorites.map((nft) => <GalleryItem nft={nft} key={nft.tokenId} />)}
+      {!loading && data && data.length > 3 && (
         <Text
           as="span"
           fontFamily="body"
           fontSize="xs"
           color="cyanText"
           cursor="pointer"
-          onClick={() => setShow(!show)}
+          onClick={onOpen}
         >
-          View {show ? 'less' : 'all'}
+          View all
         </Text>
       )}
+      <Modal
+        isOpen={isOpen}
+        onClose={onClose}
+        isCentered
+        scrollBehavior="inside"
+      >
+        <ModalOverlay css={{ backdropFilter: 'blur(8px)' }}>
+          <ModalContent maxW="75rem" bg="none">
+            <Box
+              bg="rgba(70, 20, 100, 0.8)"
+              borderTopRadius="lg"
+              p={4}
+              w="100%"
+            >
+              <HStack>
+                <Text
+                  fontFamily="mono"
+                  fontSize="sm"
+                  fontWeight="bold"
+                  color="blueLight"
+                  as="div"
+                  mr="auto"
+                >
+                  Gallery
+                </Text>
+                <FaTimes
+                  color="#A5B9F6"
```

Comment:
Should use values from the theme and also use the Icon component from Chakra: https://chakra-ui.com/docs/components/icon#using-a-third-party-icon-library

Reply:
Will replace react-icons with chakra-icons in a separate PR. Here I'm just reusing the same icon from other existing components.